src/v2/api/services/RoleService.spec.ts

```diff
@@ -0,0 +1,154 @@
+  let roleRepository: RoleRepository;
+
+  beforeEach(async () => {
+    const testingModule = await Test.createTestingModule({
+      providers: [RoleService, PermissionService],
+      imports: [PrismaModule],
+    }).compile();
+
+    roleService = testingModule.get(RoleService);
+    permissionService = testingModule.get(PermissionService);
+    roleRepository = testingModule.get(RoleRepository);
+  });
+
+  describe('canCreateGrants', () => {
+    it('should true if user has grants that is in grants array', async () => {
+      const userRoles = [{
+        grants: [{
+          permission: 'one',
+          set: true,
+        }],
+      }, {
+        grants: [{
+          permission: 'two',
+          set: true,
+        }, {
+          permission: 'three',
+          set: true,
+        }],
+      }] as DbRole[];
+      const grants = [{
+        permission: 'one',
+      }, {
+        permission: 'two',
+      }] as CreateGrantDTO[];
+
+      const result = await (roleService as any).canCreateGrants(userRoles, grants);
+
+      expect(result).toBe(true);
+    });
+
+    it('should false if user doesn\'t have at least one permission', async () => {
+      const userRoles = [{
+        grants: [{
+          permission: 'one',
+          set: true,
+        }],
+      }, {
+        grants: [{
+          permission: 'two',
+          set: true,
+        }, {
+          permission: 'three',
+          set: true,
+        }],
+      }] as DbRole[];
+      const grants = [{
+        permission: 'one',
+      }, {
+        permission: 'four',
+      }] as CreateGrantDTO[];
+
+      const result = await (roleService as any).canCreateGrants(userRoles, grants);
+
+      expect(result).toBe(false);
+    });
+  });
+
+  describe('checkRole', () => {
+    it('should throw an exception because user doesn\'t have permission on some grants', async function () {
+      const userRoles = [{
+        grants: [{
+          permission: 'roles.create',
+          set: true,
+        }],
+      }] as DbRole[];
+      const data = { grants: [{ permission: 'grants.create', set: true }] } as any;
+
+      expect.assertions(1);
+      (roleService as any).checkRole(userRoles, data)
+        .catch((e) => {
+          expect(e.message).toMatch('You do not have permission to perform this action');
+        });
+    });
+
+    it('should throw an error if parent role weight is bigger than role weight', async function () {
+      const userRoles = [{
+        grants: [{
+          permission: 'roles.create',
+          set: true,
+        }],
+      }] as DbRole[];
+      const mock = jest.spyOn(roleRepository, 'findById').mockImplementation(async (_) => {
+        return { weight: 100 } as any as Promise<DbRole>;
+      });
+      const data = { grants: [{ permission: 'roles.create', set: true }], weight: 50, parentId: 'parent' } as any;
+
+      expect.assertions(2);
+      (roleService as any).checkRole(userRoles, data)
+        .catch((e) => {
```

Comment:
Unobvious use of indention, makes call look less readable

Reply:
.catch() and .then() methods are often moved on the new line. Such approach of chaining methods also used in Builder design pattern. I don't see a problem here